c2corg_ui/static/js/documentservice.js

```diff
@@ -124,17 +124,23 @@ app.Document.prototype.hasAssociation = function(type, id) {
  * @param {appx.SimpleSearchDocument} doc
  * @param {string=} doctype Optional doctype
  * @param {boolean=} setOutingTitle
+ * @param {?boolean} editing
  * @export
  */
 app.Document.prototype.pushToAssociations = function(doc, doctype,
-    setOutingTitle) {
+    setOutingTitle, editing) {
   var associations = this.document.associations;
   doctype = doctype || app.utils.getDoctype(doc['type']);
   setOutingTitle = typeof setOutingTitle !== 'undefined' ?
     setOutingTitle : false;
   doc['new'] = true;
   associations[doctype].push(doc);
 
+  // when creating a route, make the first associated wp a main one
+  if (editing && associations.waypoints.length === 1) {
```

Comment:
I'm not sure to understand - for example, if you edit a route that has waypoints but none of them is a main one, and you only want to edit a sentence in the description of the route, it will have no impact on the waypoints.

The change occur only when you add/remove associated waypoints, or if you click on "set/unset main waypoint". Editing the route other attributes and saving them will do nothing to the associations nor to the `main_waypoint_id`.